File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"

using namespace std;

namespace raindrops {

string convert(int number)
{
    string texto = "";
    if (number % 3 == 0) texto += "Pling";
    if (number % 5 == 0) texto += "Plang";
    if (number % 7 == 0) texto += "Plong";
    if (texto == "") texto = to_string(number);

    return texto;
}

}  // namespace raindrops

File: solutions/javascript/raindrops/1/raindrops.js
//
// This is only a SKELETON file for the 'Raindrops' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const convert = (numberIntroduced) => {
  let factor3 = (numberIntroduced % 3 === 0);
  let factor5 = (numberIntroduced % 5 === 0);
  let factor7 = (numberIntroduced % 7 === 0);

  let raindrops = "";
  if (factor3) {
    raindrops += "Pling";
  }
  if (factor5) {
    raindrops += "Plang";
  }
  if (factor7) {
    raindrops += "Plong";
  }
  if (!factor3 && !factor5 && !factor7) {
    raindrops = String(numberIntroduced);
  }

  return raindrops;
};
